src/docfx/restore/Restore.cs

```diff
@@ -22,52 +45,78 @@ public static Task Run(string docsetPath, CommandLineOptions options, Report rep
                 report.Configure(docsetPath, config);
 
                 var restoredDirs = new HashSet<string>(PathUtility.PathComparer);
+                var workTreeMappings = new ConcurrentDictionary<string, Dictionary<string, string>>(PathUtility.PathComparer);
 
-                return ParallelUtility.ForEach(
-                    config.Dependencies.Values,
-                    (href, restoreChild) => RestoreDependentRepo(href, options, restoreChild, restoredDirs),
+                await ParallelUtility.ForEach(
+                    config.Dependencies.Values.Select(href => (docsetPath, href)),
+                    async (item, restoreChild) =>
+                    {
+                        var workTreeHead = await RestoreDependentRepo(item, options, restoreChild, restoredDirs);
+                        workTreeMappings.AddOrUpdate(
+                            item.docsetPath,
+                            new Dictionary<string, string> { { item.href, workTreeHead } },
+                            (key, value) =>
+                            {
+                                value[item.href] = workTreeHead;
+                                return value;
+                            });
+                    },
                     progress: Progress.Update);
+
+                if (workTreeMappings.Any())
+                {
+                    foreach (var (docset, mappings) in workTreeMappings)
+                    {
+                        if (mappings.Any())
```

Comment:
>Any [](start = 37, length = 3)

remove this Any() check, it a newer version has an empty mappings, lock file will not be updated.